docs/recipes/endpoint-testing.md

```diff
@@ -0,0 +1,23 @@
+#Endpoint Testing
+AVA doesn't have an official assertion library for endpoints, but a great option is [`supertest-as-promised`](https://github.com/WhoopInc/supertest-as-promised).
+Since the tests run concurrently, it's a best practice to create a fresh server instance for each test because if we referenced the same instance, it could be mutated between tests. This can be accomplished with a `beforeEach` and `context`, or even more simply with a factory function:
+```
+function makeApp() {
+  const app = express();
+  app.post('/signup', signupHandler);
+  return app;
+}
+```
+
+Next, just inject your server instance into supertest. The only gotcha is to use a promise or async/await syntax instead of supertest's `end` method:
+```
```

Comment:
can do, PS anyone know a way to squash commits within github without doing the pull/rebase/push dance?